### PR TITLE
[ext] Add video interpolation by param (DEV-74)

### PR DESCRIPTION
**Pull request by KonstantinKorotaev**

### CODE REVIEW

File: label_studio/data_export/api.py [hunk 1]
```diff
@@ -159,6 +159,11 @@ def get(self, request, *args, **kwargs):
         else:
             download_resources = settings.CONVERTER_DOWNLOAD_RESOURCES
 
+        if 'interpolate_key_frames' in request.GET:
+            interpolate_key_frames = bool_from_request(request.GET, 'interpolate_key_frames', False)
```
Comment (makseq) [inline label_studio/data_export/api.py:163]: Use serializer here: 
https://heartex.slack.com/archives/CKXEFJQHK/p1638175336102300

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/data_export/mixins.py [hunk 2]
```diff
@@ -126,7 +127,9 @@ def _get_export_serializer_option(self, serialization_options):
                 'annotations__completed_by'
             ].get('only_id'):
                 options['expand'].append('annotations.completed_by')
-
+            options['context'] = {'interpolate_key_frames': settings.INTERPOLATE_KEY_FRAMES}
+            if 'interpolate_key_frames' in serialization_options:
+                options['context']['interpolate_key_frames'] = serialization_options['serialization_options']
```
Comment (niklub) [inline label_studio/data_export/mixins.py:132]: serialization_options['interpolate_keyframes']

File: label_studio/data_export/serializers.py [hunk 3]
```diff
@@ -30,6 +32,18 @@ class Meta:
           'completed_by': (CompletedBySerializer, )
         }
 
+    def get_result(self, obj):
+        # run frames extraction on param, result and result type
+        if self.context.get('interpolate_key_frames', False) and obj.result and any(item['type'] == 'videorectangle' for item in list(obj.result)):
```
[the remark this answers is not in the document]
Comment (KonstantinKorotaev) [reply label_studio/data_export/serializers.py]: Fixed

File: label_studio/data_export/mixins.py (hunk 2 above)
Comment (KonstantinKorotaev) [reply label_studio/data_export/mixins.py]: Fixed

File: label_studio/data_export/api.py (hunk 1 above)
Comment (KonstantinKorotaev) [reply label_studio/data_export/api.py]: Fixed

File: label_studio/data_export/mixins.py [hunk 4]
```diff
@@ -126,7 +127,9 @@ def _get_export_serializer_option(self, serialization_options):
                 'annotations__completed_by'
             ].get('only_id'):
                 options['expand'].append('annotations.completed_by')
-
+            options['context'] = {'interpolate_key_frames': settings.INTERPOLATE_KEY_FRAMES}
+            if 'interpolate_key_frames' in serialization_options:
+                options['context']['interpolate_key_frames'] = serialization_options['serialization_options']['interpolate_key_frames']
```
Comment (niklub) [inline label_studio/data_export/mixins.py:132]: `serialization_options['interpolate_key_frames']` ?

File: label_studio/data_export/serializers.py [hunk 5]
```diff
@@ -30,6 +33,13 @@ class Meta:
           'completed_by': (CompletedBySerializer, )
         }
 
+    def get_result(self, obj):
+        # run frames extraction on param, result and result type
+        if obj.result and self.context.get('interpolate_key_frames', False) and \
+                is_video_object_tracking(obj.task.project.label_config):
```
Comment (niklub) [inline label_studio/data_export/serializers.py:39]: I don't think it's a good idea to call `parse_config` every time inside `is_video_object_tracking` - let's pass `parsed_label_config` here instead, or check for `<VideoRectangle>` substring in `label_config`

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio/data_export/serializers.py [hunk 6]
```diff
@@ -112,4 +123,19 @@ class Meta(ExportSerializer.Meta):
...
 
 
+class ExportParamSerializer(serializers.Serializer):
+    interpolate_key_frames = serializers.BooleanField(default=settings.INTERPOLATE_KEY_FRAMES,
+                                                      help_text='Interpolate video key frames.',
+                                                      required=False)
+    download_resources = serializers.BooleanField(default=settings.CONVERTER_DOWNLOAD_RESOURCES,
+                                                  help_text='Download resources in converter.',
+                                                  required=False)
+    export_type = serializers.CharField(default='JSON',
```
Comment (triklozoid) [inline label_studio/data_export/serializers.py:133]: Is it ok, that we don't support the `exportType` parameter now?

Comment (KonstantinKorotaev) [reply label_studio/data_export/serializers.py:134]: Fixed

**Review by triklozoid — APPROVED** [review]

**Review by makseq — DISMISSED** [review]

**Pull request merged**